src/ndb.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Roman Gezikov <rgezikov@exafore.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef SRC_NDB_H_
+#define SRC_NDB_H_
+
+#include <libswiftnav/common.h>
+#include <libswiftnav/ephemeris.h>
+#include <libswiftnav/almanac.h>
+#include <libswiftnav/signal.h>
+#include <ch.h>
+#include <cfs/cfs-coffee.h>
+#include <cfs/cfs.h>
+#include "signal.h"
+
+enum ndb_op_code
+{
+  NDB_ERR_NONE = 0, /**< No error */
+  NDB_ERR_MISSING_IE, /**< DB doesn't contain value of this IE */
+  NDB_ERR_UNSUPPORTED,
+  NDB_ERR_FILE_IO,
+  NDB_ERR_INIT_DONE,
+  NDB_ERR_BAD_PARAM,
+  NDB_ERR_TIMEOUT,
+  NDB_ERR_UNRELIABLE_DATA,
+  NDB_ERR_ALGORITHM_ERROR
+};
+
+enum ndb_data_source
+{
+  NDB_DS_UNDEFINED = 0,
+  NDB_DS_INIT,
+  NDB_DS_RECEIVER,
+  NDB_DS_SBP
+};
+
+#define MAX_NDB_FILE_VERSION_LEN 64
+#define ndb_file_version GIT_VERSION
+
+/** NDB IE update counter type */
+typedef u8 ndb_update_counter_t;
+/** NDB file handler */
+typedef int ndb_file_handle_t;
+/** NDB Timestamp */
+typedef systime_t ndb_timestamp_t;
+/* Information element size */
+typedef u16 ndb_ie_size_t;
+/* Information element index in the array */
+typedef u8 ndb_ie_index_t;
+
+/** NDB File */
+typedef struct __attribute__((packed)) {
+  char* name; /**< Name of the file */
+  ndb_file_handle_t fh; /**< File handle */
+  u32 expected_size; /**< Expected file size */
+} ndb_file_t;
+
+typedef union
+{
+  ephemeris_t ephe;
+  almanac_t alma;
+  u32 l2c_capabilities;
+} ndb_element_t;
```

Comment:
I think it would be better to simply allocate a buffer with the required size rather than relying on this union to be updated with every possible element type. E.g. just do `u8 buf[md->data_size]` instead of `ndb_element_t buf`.